lib/Test/Mockify/Injector.pm

```diff
@@ -0,0 +1,128 @@
+package Test::Mockify::Injector;
+use parent 'Test::Mockify::Base';
+
+use strict;
+use warnings;
+
+use Test::Mockify::Tools qw ( LoadPackage ExistsMethod );
+use Sub::Override;
+
+sub new {
+    my $class = shift;
+    my $self = $class->SUPER::new(@_);
+    $self->{'override'} = Sub::Override->new();
+    return $self;
+}
+
+sub _getFakeClass {
+    return undef;
+}
+
+sub _addMock {
+    my $self = shift;
+    my ($MethodName, $Method) = @_;
+
+    ExistsMethod( $self->_mockedModulePath(), $MethodName );
+    my $mockedSelf = $self->_mockedSelf();
+    $mockedSelf->{'__MethodCallCounter'}->addMethod( $MethodName );
+    if(not $self->{'MethodStore'}{$MethodName}){
+        $self->{'MethodStore'}{$MethodName} //= $Method;
+        my $MockedMethodBody = sub {
+            my @MockedParameters = @_;
+            $mockedSelf->{'__MethodCallCounter'}->increment( $MethodName );
+            push @{$mockedSelf->{$MethodName.'_MockifyParams'}}, \@MockedParameters;
+            return $Method->call(@MockedParameters);
+        };
+        $self->{'override'}->replace($self->_mockedModulePath().'::'.$MethodName, $MockedMethodBody);
```

Comment:
Why would you import a function into your package and then still call the fully qualified function name? I don't think this is a valid use case. Could you provide an example of why it would be?

If we assume I am correct in this thinking for a moment, you would simply mock the imported function via the package that imported it.

I get what you are saying about wanting to override the original package's implementation of a function, and all importing packages get that mocked functionality for free. My hesitation is that I don't think there is a way to achieve this without specifying both packages to the Injector, and that seems really clunky. That was essentially how you achieved it when you first overhauled my changes.

I think this is probably an edge case, anyway. Do you feel it is really important to address?

Reply:
I have the impression we did't talk about the same thing. I will have now a look again into the PR. Lets see what I misunderstood.

Reply:
ok, I think now i get it. The thing I didn't realized was this:
```perl
my $injectorImported = Test::Mockify::Injector->new('FakeModuleStaticInjection');
    $injectorImported->mock('ReturnHelloWorld')->when(String('German'))->thenReturn('Hallo Welt');
```
I didn't realize that the injector doesn't create a 'FakeStaticTools' but a 'FakeModuleStaticInjection'. And than there is mock a for a Function which is not owned by this Package. This sounds wrong to me. Off course when using "use name qw (Function)" this Function will become part of this module and the "module->can()" check is true. But this is in-transparent. In my company we are using only this import option. This would result in unmaintainable test code. 
For the full qualified Call your solution is fine but not for imported functions.
I think this is the reason for the two different views on this injecting problem, namings, aso.  I will rethink the whole problem. and let you know.